src/main/java/seedu/address/logic/commands/GroupCommand.java

```diff
@@ -52,12 +57,40 @@ public CommandResult execute(Model model) throws CommandException {
         if (model.hasGroupName(new Group(groupName, List.of()))) {
             throw new CommandException(MESSAGE_DUPLICATE_GROUP);
         }
+
         model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
         List<Person> allPersons = model.getFilteredPersonList();
-        List<Person> groupMembers = allPersons.stream()
-                .filter(person -> students.contains(person
-                        .getName().fullName))
-                .toList();
+        List<Person> groupMembers = new ArrayList<>();
+        List<String> notFoundStudents = new ArrayList<>();
+
+        Set<String> uniqueStudents = new HashSet<>();
+        for (String studentName : students) {
+            if (studentName.equals("")) {
```

Comment:
Good job error checking, seems like you got the hang of software engineering principles